### PR TITLE
gnrc/lwmac : Expose configurations to Kconfig

### PR DESCRIPTION
**Pull request by akshaim**
### Contribution description

This PR exposes compile configurations in GNRC: LWMAC to Kconfig.

### Testing procedure

1. New documentation was built using Doxygen 

The build works fine.

2. Header and macro was introduced to tests/gnrc_lwmac/

```

#include "net/gnrc/lwmac/lwmac.h"
#include "net/gnrc/lwmac/types.h"

#define STR(x)   #x
#define SHOW_DEFINE(x) printf("%s=%s\n", #x, STR(x))

```
The test file can be found [here](https://github.com/akshaim/RIOT/commit/e2d42ba4e39fddcf4b419917661a0d8b85ce49a1) .
Compiled binaries were uploaded to FIT IoT Test bed.
##### Firmware Output

main(): This is RIOT! (Version: 2020.07-devel-717-gb00e4-Kconfig_lwmac_tests)
CONFIG_GNRC_LWMAC_WAKEUP_INTERVAL_US=(200LU *(1000LU))
CONFIG_GNRC_LWMAC_TIME_BETWEEN_WR_US=(5U *(1000LU))
CONFIG_GNRC_LWMAC_WR_PREPARATION_US=((3U *(1000LU)))
CONFIG_GNRC_LWMAC_DATA_DELAY_US=(10U *(1000LU))
CONFIG_GNRC_LWMAC_DATA_CSMA_RETRIES=(3U)
CONFIG_GNRC_LWMAC_MAX_DATA_TX_RETRIES=(3U)
CONFIG_GNRC_LWMAC_MAX_RX_EXTENSION_NUM=(3U)
CONFIG_GNRC_LWMAC_BROADCAST_CSMA_RETRIES=(3U)
CONFIG_GNRC_LWMAC_TIMEOUT_COUNT=(3U)
LWMAC test application

#### Usage with menuconfig [default values]

> make menuconfig

##### Firmware Output

main(): This is RIOT! (Version: 2020.07-devel-717-gb00e4-Kconfig_lwmac_tests)
CONFIG_GNRC_LWMAC_WAKEUP_INTERVAL_US=200000
CONFIG_GNRC_LWMAC_TIME_BETWEEN_WR_US=5000
CONFIG_GNRC_LWMAC_WR_PREPARATION_US=3000
CONFIG_GNRC_LWMAC_DATA_DELAY_US=10000
CONFIG_GNRC_LWMAC_DATA_CSMA_RETRIES=3
CONFIG_GNRC_LWMAC_MAX_DATA_TX_RETRIES=3
CONFIG_GNRC_LWMAC_MAX_RX_EXTENSION_NUM=3
CONFIG_GNRC_LWMAC_BROADCAST_CSMA_RETRIES=3
CONFIG_GNRC_LWMAC_TIMEOUT_COUNT=3
LWMAC test application

#### Usage with menuconfig

> make menuconfig

##### Firmware Output

main(): This is RIOT! (Version: 2020.07-devel-717-gb00e4-Kconfig_lwmac_tests)
CONFIG_GNRC_LWMAC_WAKEUP_INTERVAL_US=300000
CONFIG_GNRC_LWMAC_TIME_BETWEEN_WR_US=6000
CONFIG_GNRC_LWMAC_WR_PREPARATION_US=2000
CONFIG_GNRC_LWMAC_DATA_DELAY_US=15000
CONFIG_GNRC_LWMAC_DATA_CSMA_RETRIES=2
CONFIG_GNRC_LWMAC_MAX_DATA_TX_RETRIES=4
CONFIG_GNRC_LWMAC_MAX_RX_EXTENSION_NUM=4
CONFIG_GNRC_LWMAC_BROADCAST_CSMA_RETRIES=2
CONFIG_GNRC_LWMAC_TIMEOUT_COUNT=4
LWMAC test application

**MACROS were successfully configured.**

### Issues/PRs references

#12888 
~~Merge this only after merging #14086~~

### CODE REVIEW

File: sys/include/net/gnrc/lwmac/types.h [hunk 1]
```diff
@@ -90,12 +90,18 @@ extern "C" {
...
+ * @ingroup  net_gnrc_conf
+ * @{
+ */
 /**
  * @brief The default largest number of parallel timeouts in LWMAC
  */
 #ifndef GNRC_LWMAC_TIMEOUT_COUNT
 #define GNRC_LWMAC_TIMEOUT_COUNT             (3U)
 #endif
+/** @} */
```
Comment (leandrolanzieri) [inline sys/include/net/gnrc/lwmac/types.h]:
```suggestion
/**
 * @ingroup net_gnrc_lwmac_conf
 * @brief The default largest number of parallel timeouts in LWMAC
 */
#ifndef GNRC_LWMAC_TIMEOUT_COUNT
#define GNRC_LWMAC_TIMEOUT_COUNT             (3U)
#endif
```
That should be enough to add the macro to the group.

File: sys/net/gnrc/link_layer/lwmac/lwmac.c [hunk 2]
```diff
@@ -293,14 +293,15 @@ void lwmac_set_state(gnrc_netif_t *netif, gnrc_lwmac_state_t newstate)
                 uint32_t alarm;
 
                 rtt_clear_alarm();
-                alarm = random_uint32_range(RTT_US_TO_TICKS((3 * GNRC_LWMAC_WAKEUP_DURATION_US / 2)),
-                                            RTT_US_TO_TICKS(GNRC_LWMAC_WAKEUP_INTERVAL_US -
-                                                            (3 * GNRC_LWMAC_WAKEUP_DURATION_US / 2)));
+                alarm = random_uint32_range(RTT_US_TO_TICKS((3 *
```
Comment (leandrolanzieri) [inline sys/net/gnrc/link_layer/lwmac/lwmac.c:296]: Please keep the original indentation and wrapping when possible. Also drop unnecessary `\`.

File: sys/net/gnrc/link_layer/lwmac/rx_state_machine.c [hunk 3]
```diff
@@ -143,8 +143,9 @@ static bool _send_wa(gnrc_netif_t *netif)
                                    _gnrc_lwmac_ticks_to_phase(netif->mac.prot.lwmac.last_wakeup));
     }
     else {
-        lwmac_hdr.current_phase = (phase_now + RTT_US_TO_TICKS(GNRC_LWMAC_WAKEUP_INTERVAL_US)) -
-                                  _gnrc_lwmac_ticks_to_phase(netif->mac.prot.lwmac.last_wakeup);
+        lwmac_hdr.current_phase = (phase_now + \
```
Comment (leandrolanzieri) [inline sys/net/gnrc/link_layer/lwmac/rx_state_machine.c:146]: `\` not needed here and in other places.

File: sys/net/gnrc/link_layer/lwmac/Kconfig [hunk 4]
```diff
@@ -0,0 +1,142 @@
+# Copyright (c) 2020 Freie Universitaet Berlin
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+menuconfig KCONFIG_MODULE_GNRC_LWMAC
+    bool "Configure GNRC_LWMAC"
```
Comment (leandrolanzieri) [inline sys/net/gnrc/link_layer/lwmac/Kconfig:8]:
```suggestion
    bool "Configure GNRC LWMAC"
```

File: sys/net/gnrc/link_layer/lwmac/Kconfig [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+    depends on MODULE_GNRC_LWMAC
+    help
+        Configure the GNRC_LWMAC using Kconfig.
```
Comment (leandrolanzieri) [inline sys/net/gnrc/link_layer/lwmac/Kconfig:11]:
```suggestion
        Configure the GNRC LWMAC module using Kconfig.
```

File: sys/net/gnrc/link_layer/lwmac/lwmac.c (hunk 2 above)
Comment (akshaim) [reply sys/net/gnrc/link_layer/lwmac/lwmac.c:296]: Yes. This was an old commit. Will follow the orignal coding style from now on.

**Comment by leandrolanzieri** [comment]
This needs rebasing

**Comment by leandrolanzieri** [comment]
> This needs rebasing

And again

**Comment by leandrolanzieri** [comment]
Looks good. Please squash

**Comment by leandrolanzieri** [comment]
@akshaim you'll need to rebase this

**Comment by akshaim** [comment]
> @akshaim you'll need to rebase this

Done.

**Pull request merged**